src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -220,12 +218,18 @@ protected function prepareJsonResponse($request, Exception $e)
 
         $headers = $this->isHttpException($e) ? $e->getHeaders() : [];
 
-        return response(json_encode([
-            'message' => $e->getMessage(),
-            'file' => $e->getFile(),
-            'line' => $e->getLine(),
-            'trace' => $e->getTrace(),
-        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
+        if (config('app.debug')) {
+            $response = [
+                'message' => $e->getMessage(),
+                'file' => $e->getFile(),
+                'line' => $e->getLine(),
+                'trace' => $e->getTrace(),
```

Comment:
Sorry, I don't think we should response a different json in debug mode and the production mode. 

Further more, the exception code (getCode()) may have much more valuable.

May be we should just hide the trace for some security consideration. Also we should consider the translation for the message.